### PR TITLE
remove Cstruct dependency

### PR DESCRIPTION
**Pull request by palainp**
Dear devs,
I came here while on the path to removing `Cstruct.t` to avoid small bigarray allocations (which don't seem to help GC), here it's only a 16B header per xenstore message.
Removing `ppx_cstruct` forced me defining header fields offsets manually but so far `dune runtest` passes and qubes-mirage-firewall works as intended. The `Buffer.t` type is still there as I didn't know how to write `Buffer.add_string x.data bytes;` with `Bytes.t`.
With this PR I don't expect any performance improvement (nor degradation), but simply lighten the dependency cone :)

### CODE REVIEW

**Comment by palainp** [comment]
CI reports `Bytes.set_int32_le` being introduced in Ocaml 4.08, could it be ok to drop older compiler support?
Ocaml 5+ seems to be unrelated to that PR.

File: core/xs_protocol.ml [hunk 1]
```diff
@@ -149,29 +149,33 @@ type t = {
...
-  set_header_rid header pkt.rid;
-  set_header_tid header pkt.tid;
-  set_header_len header len;
-  let result = Buffer.create 64 in
-  Buffer.add_bytes result (Cstruct.to_bytes header);
-  Buffer.add_buffer result pkt.data;
-  Buffer.to_bytes result
+  let len = Buffer.length pkt.data in
+  let result = Bytes.create (sizeof_header + len) in
+  set_header_ty result @@ Op.to_int32 pkt.ty ;
```
Comment (hannesm) [inline core/xs_protocol.ml:173]:
```suggestion
  set_header_ty result (Op.to_int32 pkt.ty) ;
```

File: core/dune [hunk 2]
```diff
@@ -2,7 +2,4 @@
  (name xenstore)
  (public_name xenstore)
  (wrapped false)
- (libraries cstruct)
- (preprocess
-  (pps ppx_cstruct))
  (flags :standard -safe-string))
```
Comment (hannesm) [inline core/dune:5]:
```suggestion
)
```

File: core/dune [hunk 3]
```diff
@@ -2,7 +2,4 @@
  (name xenstore)
  (public_name xenstore)
  (wrapped false)
```
Comment (hannesm) [inline core/dune:4]:
```suggestion
 (wrapped false))
```

File: core/dune [hunk 4]
```diff
@@ -2,7 +2,4 @@
...
- (libraries cstruct)
- (preprocess
-  (pps ppx_cstruct))
- (flags :standard -safe-string))
+)
```
Comment (hannesm) [inline core/dune:5]:
```suggestion
```

**Comment by hannesm** [comment]
Thanks, I pushed some minor changes. If CI is fine, I'll merge and release.

**Comment by palainp** [comment]
Thank you @hannesm!

**Pull request merged**